keystore/file_keystore.go

```diff
@@ -362,7 +355,7 @@ func (k *FileKeystore) encrypt(reader io.Reader) (io.Reader, error) {
 		return nil, fmt.Errorf("could not create the keystore cipher to encrypt, error: %w", err)
 	}
 
-	data, err := ioutil.ReadAll(reader)
+	data, err := io.ReadAll(reader)
```

Comment:
Why not `iobuf`?

Reply:
That was the idea initially, but the io.Copy is not always the best approach, please see https://github.com/elastic/beats/issues/36151#issuecomment-2362227748. I don't feel confortable changing this code without having proper benchmarks in place to confirm it is indeed faster. If you think this is still worth doing, I can work on some benchmarks and replace with iobuf.